### ISSUE TITLE
[BUG] Requests with invalid page index time-out

### ISSUE THREAD

**Issue by psla** [paperless-ngx/paperless-ngx#3079]
### Description

*I found this bug when investigating https://github.com/paperless-ngx/paperless-ngx/issues/3077; I was also tracking what I thought was "hangs" when importing new docs - now I believe it was not hangs, it was just this error*

# Problem 1 - requests with invalid page index time-out

The following get executes correctly:

```
https://paperless.usa.sepio.pl/api/documents/?page=1&page_size=50&ordering=-created&truncate_content=true&title_content=a
```

However, replace the `page=1` with an index larger than maximum page (e.g. 100 in my case), and the request will never finish.

# Problem 2 - search box does not reset page index when searching.

Navigate to the documents page. Navigate to page number 8. Search for a unique keyward. The requested URL has `page=8` attached, instead of `page=1` (for a filter/search, that original pagination doesn't longer apply really).

### Steps to reproduce

An easy and consistent repro is

1) Create a document with a unique word (`superunique`)
2) Navigate to the documents page. You must have more than one page in the documents page.
3) Click on page larger than one. (in my case, page 8): Here is how it looks on mine
<img width="863" alt="image" src="https://user-images.githubusercontent.com/161429/231350076-78682fd7-9190-493d-84d6-cbc696f8667a.png">

4) Search for `superunique`.
5) Observe: the spinning wheel never finishes / the search never finishes (it actually timeouts depending on settings):
<img width="250" alt="image" src="https://user-images.githubusercontent.com/161429/231350314-b2f558e6-8b6c-4ab7-88bc-66d6c29bd1f4.png">

This is because the request attaches `page=8` to the request, instead of using `page=1`.

Observed behavior: In my case I eventually get proxy timeout
<img width="877" alt="image" src="https://user-images.githubusercontent.com/161429/231350560-fdedb61d-28bc-4efd-b57a-a5d3895872ad.png">

### Webserver logs

```bash
Example for search string 'abc':


access-paperless.log:10.0.7.76 - - [11/Apr/2023:21:29:17 -0700] "GET /api/documents/?page=1&page_size=50&ordering=-created&truncate_content=true&title_content=abc HTTP/1.1" 401 5592 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/111.0"
access-paperless.log:10.0.7.76 - piotr [11/Apr/2023:21:28:42 -0700] "GET /api/documents/?page=100&page_size=50&ordering=-created&truncate_content=true&title_content=abc HTTP/1.1" 502 5498 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/111.0"


THe first request (`page=1`) succeeded (200), second request `page=100` failed (502)
```

### Paperless-ngx version

Paperless-ngx 1.13.0

### Host OS

5.10.0-21-amd64 #1 SMP Debian 5.10.162-1 (2023-01-21) x86_64 GNU/Linux

### Installation method

Docker - official image

### Browser

Reproducible in Chrome & Firefox

**Comment by psla** [comment]
Thanks for the quick response and apologies for filing a dupe!

The repro for me is extremely trivial - all I need to do is to paste the link (`https://myhost/api/documents/?page=5&page_size=50&ordering=-created&truncate_content=true&title_content=abc`) in the address bar and then the request never ends. I'll dig deeper into that :)

**Comment by shamoon** [comment]
No worries, yea it seems like if theres an issue its somewhere Django not properly returning a 404, clearly that example should just 404 (and for me it does)

```
curl -i 'http://localhost:8000/api/documents/?page=5&page_size=50&ordering=-created&truncate_content=true&title_content=abc' -u 'admin:admin'
HTTP/1.1 404 Not Found
Content-Type: application/json
Vary: Accept, Accept-Language, Origin
Allow: GET, POST, HEAD, OPTIONS
X-Frame-Options: ANY
X-Api-Version: 2
X-Version: 1.14.0
Content-Length: 26
Content-Language: en-us
X-Content-Type-Options: nosniff
Referrer-Policy: same-origin
Cross-Origin-Opener-Policy: same-origin

{"detail":"Invalid page."}%
```


https://user-images.githubusercontent.com/4887959/231353153-ab48a64d-3d3f-487a-8507-7ccef8e876db.mov

**Comment by psla** [comment]
yup, thanks. Not too familiar with django, but I'll try to spend some time figuring it out.

I totally believe you. To remove my proxy from the picture (apache reverse proxy), I logged in to the container itself (official) and run the same command:

```
c# curl -i 'http://localhost:8000/api/documents/?page=1&page_size=50&ordering=-created&truncate_content=true&title_content=abc' -u 'user:pass'
HTTP/1.1 200 OK
date: Wed, 12 Apr 2023 04:54:33 GMT
server: uvicorn
content-type: application/json
vary: Accept, Accept-Language, Origin, Accept-Encoding
allow: GET, HEAD, OPTIONS
x-frame-options: SAMEORIGIN
x-api-version: 2
x-version: 1.13.0
content-length: 4277
content-language: en-us
x-content-type-options: nosniff
referrer-policy: same-origin
cross-origin-opener-policy: same-origin

{"count":4,"next":null,"previous":null,"results":[{"id":55,"correspondent":null,"document_type":null,"storage_path":null,"title":"23ddab....... {{FULL JSON RESPONSE HERE }}

root@d3ebb471ef1c:/usr/src/paperless/src# curl -i 'http://localhost:8000/api/documents/?page=2&page_size=50&ordering=-created&truncate_content=true&title_content=abc' -u 'user:pass'

### THIS NEVER ENDS
```

So weird -- I would totally expect it to work the same in docker...
